lib/ddtrace/workers/polling.rb

```diff
@@ -26,8 +26,12 @@ def stop(force_stop = false, timeout = SHUTDOWN_TIMEOUT)
           stop_loop
           graceful = join(timeout)
 
-          # If timeout and force stop...
-          !graceful && force_stop ? terminate : graceful
+          if !graceful && force_stop
+            Datadog.logger.debug "Timeout while waiting for worker to finish gracefully, forcing termination for: #{self}"
```

Comment:
General question: is there a policy in place for `logger.debug "..."` vs `logger.debug { "..." }`?

(the latter does not execute the block, hence while it creates a block, it never allocates/processes the string if `logger.debug?` is falsey)

Reply:
Also (nitpick) style is inconsistent with the other calls, which use `.debug("...")`

Reply:
> General question: is there a policy in place for logger.debug "..." vs logger.debug { "..." }?

Good question! I don't think so -- grepping the codebase shows that most debug logging is using the non-block approach.

_In these specific cases_ I don't expect them to happen so often that the debug logging would impact performance -- e.g. if they happen often, then there's probably something bigger wrong than just a bit of extra work for a string.

BUT in general just using the block version is simpler than me speculating on frequency of debug information so I've switched both of the strings using interpolation to the block version in 07d9b8a794091628b44e408f6d7b7c669c75e518 .